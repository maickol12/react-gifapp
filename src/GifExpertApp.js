import React, { useState } from 'react';
import CategoryAdd from './components/CategoryAdd';
import { CategoryGifGrid } from './components/CategoryGifGrid';

const GifExpertApp = ({ defaultCategories = []}) => {
    //const categories = ['One Punch','Samurai x','Dragon ball']
    const [categories, setCategory] = useState(defaultCategories);
    const handleAdd = (categoryName) => {
         setCategory([...categories,categoryName]);
    }
    return(
        <div>
            <h2>GifExpertApp</h2>
            <hr />
            <CategoryAdd setCategory={handleAdd}  />
            {
                <ol>
                    { categories.map((value,index) => <CategoryGifGrid key={`cat${index}`} category={value} /> ) }
                </ol>
            }
        </div>
    )
}

export default GifExpertApp;