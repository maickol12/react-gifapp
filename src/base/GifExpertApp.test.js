import { shallow } from "enzyme";
import GifExpertApp from "../GifExpertApp";

describe('Probando <GifExpertApp />',() => {
    const wrapper = shallow(<GifExpertApp />);
    test('Debe mostrarse correctamente',() => {
        expect( wrapper ).toMatchSnapshot(); 
    });
    test('debe de mostrar una lista de categorias', () => {
        const categories = ['Mamberroi','One Punch'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('CategoryGifGrid').length ).toBe( categories.length );
    })
});
