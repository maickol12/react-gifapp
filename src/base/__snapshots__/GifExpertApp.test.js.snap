// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando <GifExpertApp /> Debe mostrarse correctamente 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <CategoryAdd
    setCategory={[Function]}
  />
  <ol />
</div>
`;

exports[`Probando <GifExpertApp /> debe de mostrar una lista de categorias 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <CategoryAdd
    setCategory={[Function]}
  />
  <ol>
    <CategoryGifGrid
      category="Mamberroi"
      key="cat0"
    />
    <CategoryGifGrid
      category="One Punch"
      key="cat1"
    />
  </ol>
</div>
`;
